src/cpp/client.cpp

```diff
@@ -797,7 +818,7 @@ inline void Client::_append_with_put_prefix(
 
 inline CommandReply Client::_get_dataset_metadata(const std::string& name)
 {
-    Command cmd;
+    CompoundCommand cmd;
```

Comment:
In general putting and getting a ``DataSet`` is a compound Command, but the ``Client`` current breaks it up into pieces, and this ``HGETALL`` is actually a ``SingleKeyCommand``